src/WebJobs.Extensions.DurableTask.CodeGeneration/README.md

```diff
@@ -0,0 +1,210 @@
+    mockOrchestrationContext.Verify(c => c.CallActivityAsync<int>("Add", new Tuple<int, int>(num1, num2)), Times.Once);
+}
+```
+
+## The Solution
+
+Using the recently released resource Roslyn-based [Source Generators](https://devblogs.microsoft.com/dotnet/introducing-c-source-generators/) we can automatically generate code to abstract the orchestration/activity call. 
+
+The function name, duable function type (orchestration/activity), parameters (names/types) and return type call can all be discovered by statically analyzing the code. 
+
+An ```IGeneratedDurableOrchestrationCaller``` is generated for representing all Orchestration function calls in the project.
+
+An ```IGeneratedDurableActivityCaller``` is generated for representing all Activity function calls in the project.
+
+An ```IGeneratedDurableOrchestrationStarter``` is generated for representing all Orchestration function starts in the project.
+
+An ```IGeneratedDurableOrchestrationContext``` interface is generated representing an interface for wrapping all operations against previous usage of ```IDurableOrchestrationContext``` and supporting calling Orchestration/Activity functions in the project - exposing an ```IGeneratedDurableOrchestrationCaller``` / ```IGeneratedDurableActivityCaller```.
+
+An ```IGeneratedDurableClient``` interface is generated representing an interface for wrapping all operations against previous usage of ```IDurableClient``` and supporting starting Orchestration functions in the project - exposing ```IGeneratedDurableOrchestrationStarter```.
+
+Paired concrete implementations of the preceding interfaces are also generated to automatically handle forwarding all the calls appropriate through the pre-existing operations exposed by the DurableTask framework.
+
+```csharp
+public partial interface IGeneratedDurableActivityCaller
+{
+    /// <summary>
+    /// See <see cref = "WebJobs.Extensions.DurableTask.CodeGen.Example.Calculator.Add"/>
+    /// </summary>
+    Task<int> Add(int num1, int num2);
+}
+```
+
+```csharp
+public partial interface IGeneratedDurableActivityCaller
+{
+    /// <summary>
+    /// See <see cref = "WebJobs.Extensions.DurableTask.CodeGen.Example.Calculator.Add"/>
+    /// </summary>
+    Task<int> Add(int num1, int num2) 
+    {
+        return context.CallActivityAsync<int>("Add", (num1, num2));
+    }
+}
+```
+
+```csharp
+[FunctionName("Multiply")]
+public async Task<int> Multiply(
+    [OrchestrationTrigger] IGeneratedDurableOrchestrationContext context
+)
+{
+    var (num1, num2) = context.GetInput<(int, int)>();
+
+    var result = 0;
+
+    for (var i = 0; i < num2; i++)
+    {
+        result = await context.Activity.Add(result, num1);
+    }
+
+    return result;
+}
+```
+
+## Limitations
+
+### 1. Manually Adding Namespace
+
+Generated code is placed in the namespace ```Microsoft.Azure.WebJobs.Generated```. For the moment, this will not show up in intellisense. To use the generated interfaces, you must manually include 
```

Comment:
Looks like this sentence needs to be finished?

Reply:
Right. Updated the section.